### ISSUE TITLE
update to latest cda logical model 2.0.0-sd-snapshot1

### ISSUE THREAD

**Issue by oliveregger** [ahdis/matchbox#196]
HL7 has published a new version by 2023.12

https://hl7.org/cda/stds/core/2.0.0-sd-snapshot1/

we should update our logical model based on the 2.0.0 version.

**Comment by oliveregger** [comment]
the latest model version using the tooling extensions which have been integrated only in the core libraries in 6.2.5, we are using currently 6.1.2.2., need to update the core libraries

https://github.com/hapifhir/org.hl7.fhir.core/blame/master/org.hl7.fhir.r5/src/main/java/org/hl7/fhir/r5/elementmodel/ParserBase.java

**Comment by oliveregger** [comment]
The model update has braking changes and existing FML maps need to be updated to work:

- where there is a xs:choice group the models introduces a new item element, where the maps need to be added, see EN, AD mappings: see also https://github.com/ahdis/matchbox/blob/9ca1d9dadd08a0f67b4d035d11b9542f83407811/matchbox-engine/src/test/resources/cda/datatypes.map

- for sdtc extensions, the element names have been prefixed now with sdtc, e,g.

//		assertEquals("active", getEngine().evaluateFhirPath(cdaLabItaly, false, "statusCode.code"));
		// changed with core 7.3.0
    assertEquals("active", getEngine().evaluateFhirPath(cdaLabItaly, false, "sdtcStatusCode.code"));

**Comment by robeastwood-agency** [comment]
Thank you very much for those updates; looks good. And I also came to the same conclusion regarding these breaking changes, particularly with the new `item` element (we don't use sdtc extensions).

Regarding the new `item` element, I have updated the relevant FML maps e.g. for address datatype, modelled after your work in the HL7CH mapping repo, and have had to do this to enable a transform to occur (ie to avoid errors being raised):
```
group AddressAD(source src : Address, target tgt : AD) extends Any <<types>> {

  src -> tgt.item as item then {

    src.line as v -> item.streetAddressLine as line, line.xmlText = v "streetAddress";
    src.postalCode as v -> item.postalCode as postalCode, postalCode.xmlText = v "postalCode";
    src.city as v -> item.city as city, city.xmlText = v "city";
    src.state as v -> item.state as state, state.xmlText = v "state";
    src.district as v -> item.county as county, county.xmlText = v "county";
    src.country as v -> item.country as country, country.xmlText = v "country";    

  } "addritem";
}
```
However, when the transform returns a CDA document from a FHIR Bundle, instances of address all look like this, ie with the `item` element:
```
<addr>
    <item>
        <country>Australia</country>
        <state>NSW</state>
        <city>Nehtaville</city>
        <postalCode>5555</postalCode>
        <streetAddressLine>1 Clinician Street</streetAddressLine>
    </item>
</addr>
```
I cannot figure out how to readjust the FML to suppress the `item` element in order to pass schema validation. I am wondering if there is something I can do or has it got something to do with the base CDA logical model, with how `item` is defined. 

Note that instances of person name behave in the same way as address.

Any clues? thank you

**Comment by oliveregger** [comment]
@robeastwood-agency i think the model should have an attribute that item does not get serialized (which would then also to be respected by the serializer). I will have a look into it how we can support that with matchbox in the next version.

Would it be possible to provide a PR for the updated maps in the  HL7CH mapping repo?

**Comment by robeastwood-agency** [comment]
OK thanks @oliveregger 

Yes I'll prepare a PR for [hl7ch/cda-fhir-maps](https://github.com/hl7ch/cda-fhir-maps) to update the relevant maps

**Comment by oliveregger** [comment]
the model has the following extension on the item elements which are in the choice group:

```xml
      <extension url="http://hl7.org/fhir/tools/StructureDefinition/xml-choice-group">
        <valueBoolean value="true"/>
      </extension>
```

**Comment by robeastwood-agency** [comment]
We also noticed that extension, and furthermore its definition stating that "_this element is a choice group that doesn’t literally appear in the XML_" (http://hl7.org/fhir/tools/0.1.0/StructureDefinition-xml-choice-group.html).

On a related piece of work on our CDA specifications derived from the core CDA logical model, we have an example CDA instance being processed by the IG Publisher and the generated version of it also has the `item` element inserted for address and name.

 It seems then to me that as the extension is present, it may be that the serializer is not respecting that it should not appear in instances as you suggest.

I'll raise a discussion on zulip to pursue.

Edit: zulip thread: https://chat.fhir.org/#narrow/stream/246127-CDA-IG-Publisher/topic/CDA.20Logical.20Model.20-.20Choice.20Group.20.60item.60.20is.20not.20suppressed

**Comment by robeastwood-agency** [comment]
As a result of the zulip thread, Grahame G asked me to create a bug ticket on the core library; see here https://github.com/hapifhir/org.hl7.fhir.core/issues/1583

**Comment by oliveregger** [comment]
we will proivde either a PR or use the fix in matchbox, whoever is faster :-)

**Comment by oliveregger** [comment]
fixed in version 3.8.1
